### PR TITLE
[SPARK-15875] Try to use Seq.isEmpty and Seq.nonEmpty instead of Seq.length == 0 and Seq.length > 0

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

In scala, immutable.List.length is an expensive operation so we should
avoid using Seq.length == 0 or Seq.lenth > 0, and use Seq.isEmpty and Seq.nonEmpty instead.
## How was this patch tested?

existing tests

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 1]
```diff
@@ -1656,7 +1656,7 @@ class Analyzer(
 
         // We do a final check and see if we only have a single Window Spec defined in an
         // expressions.
-        if (distinctWindowSpec.length == 0 ) {
+        if (distinctWindowSpec.isEmpty ) {
```
Comment (srowen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: LGTM. Tiny nit, if you have to make another edit: remove space inside paren

Comment (yangw1234) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: @srowen fixed it.

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test.

**Comment by rxin** [comment]
Merging in master/2.0. I'm not sure whether any of these cases are actually perf sensitive though. If they are, they probably shouldn't be using Seq anyway.

**Pull request closed without merging**